### ISSUE TITLE
How to work with JWT server-to-server auth?

### ISSUE THREAD

**Issue by mercteil** [googleapis/google-api-nodejs-client#1793]
I have a microservice worker which reads files from google drive and monitors activity for a particular folder. 

Therefore I use a service account with privateKey credentials and authorize with a JWT Object.

My problem is the work with the JWT Token. I expect that there is a way to check whether any token is expired or not and my service worker can do a refresh prior firing any requests to googleapis. But it seems there is no such option. I had an issue that when I define the JWT Object once at some point my worker throws an Error and it seems to come from JWT refreshing, which kind of happens automatically?

I now ended up requesting a new JWT Object on every request, which on the other hand leads to unnecessary multiplication of the overall requests amount. 

My code below defines JWTOptions and method to get a fresh JWT Object, which is executed on every request.

Any suggestions how I can define the JWT token and check its expiration date to be able to refresh the JWT only when it expires instead of requesting a new one before every googleapi call.

```typescript
...

private JWTOptions: JWTOptions = {
    email: config.google.clientEmail,
    key: config.google.privateKey,
    scopes: config.google.scopes,
  };

  private refreshJWT = (): JWT => new google.auth.JWT(this.JWTOptions);

  private drive = (): IDrive.Drive => {
    return google.drive({ version: 'v3', auth: this.refreshJWT() });
  };

  private driveActivity = (): IDriveActivity.Driveactivity => {
    return google.driveactivity({ version: 'v2', auth: this.refreshJWT() });
  };

  fetchSingleImageById(fileId: string): Promise<ArrayBuffer> {
    return this.drive()
      .files.get({ fileId, alt: 'media' }, { responseType: 'arraybuffer' })
      .then((res): any => {
        const { data } = res;
        return data;
      });
  }

...
```

**Comment by JustinBeckwith** [comment]
Greetings!  The `google.auth.JWT` class does the refreshing for you automatically :) We look at the expiry_date of the returned token, and then automatically refresh during a request if we think the token is (or is about to be expired).  

TL;DR: this should just work :)

**Comment by mercteil** [comment]
It works automatically, but once in a while I received a gaxios error, which is fatal for my application.

This is why I wanted to do it manually.

**Comment by JustinBeckwith** [comment]
Can you share the error you get?

**Comment by JustinBeckwith** [comment]
We've fixed a bunch of issues in the automatic token refreshing logic recently, and I'd wager this has been fixed.  Let us know if you're still running into problems!
